PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -331,13 +331,13 @@ public function write($data)
     public function error_handler($errno, $errstr, $errfile, $errline, $errcontext = null)
     {
         // fwrite notice that the stream isn't ready - EAGAIN or EWOULDBLOCK
-        if ($errno == SOCKET_EAGAIN || $errno == SOCKET_EWOULDBLOCK) {
+        if (strpos($errstr, 'errno=' . SOCKET_EAGAIN) !== false || strpos($errstr, 'errno=' . SOCKET_EWOULDBLOCK) !== false) {
```

Comment:
I would extract the error code one with a regex, instead of checking the string three times.

Reply:
I've adapted to code to use a regular expression.

Reply:
The problem of course is that now we're searching strings again. @marek-obuchowicz - could you please check your non-English environment and provide an example of the `errstr`? I am hoping that `errno=11` is still part of it.

Reply:
One more thing - I prefer the use of `strpos` as opposed to a regex. Please "pre-compute" the substrings to search and save them as static consts in the `StreamIO` class itself.

Reply:
The error message with german locale set is following:
`fwrite(): send of 49 bytes failed with errno=11 Die Ressource ist zur Zeit nicht verfügbar`

Reply:
The error message with german locale set is following:
fwrite(): send of 49 bytes failed with errno=11 Die Ressource ist zur Zeit nicht verfügbar

Reply:
Thank you!

Reply:
I've reverted the changes for using a regular expression and pre-computed the strings (as private static attributes of `StreamIO`).